### ISSUE TITLE
Incorrect Timezone Sent to Guests in Different Countries When Booking a Meeting

### ISSUE THREAD

**Issue by AlleyNawaz** [calcom/cal.com#17033]
### Is your proposal related to a problem?

I’ve noticed an issue when scheduling meetings with guests in different countries. When I book a meeting and add guests, the timezone displayed in the meeting invite is based on my timezone, and this same timezone is sent to all guests. This can cause confusion for guests located in other time zones, as the meeting time may not match their local time.

### Describe the solution you'd like

I propose adding an option that prompts the organizer to check whether the guest is from the same country after they are added. If the guest is from a different country, the organizer should be able to select the guest’s timezone or country, so the correct time is displayed for them in the invite.

This will ensure that the meeting time sent to the guests reflects their local timezone, avoiding any misunderstandings about the meeting schedule.

### Describe alternatives you've considered

One alternative could be to automatically detect the guest’s timezone based on their email address or location, but this might not always be accurate. Another solution could be to simply provide a warning to the organizer, reminding them to double-check the time zone when inviting international guests.

### Additional context

This feature will significantly help teams working across different time zones, ensuring meetings are scheduled correctly for all participants, no matter where they are located.

### Requirement/Document

There could be documentation explaining how to use the feature and ensuring users are aware of the timezone setting options while adding guests.

---

##### House rules

- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.
  - Follow Best Practices lined out in our [Contributor Docs](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)

**Comment by Praashh** [comment]
can you attach, some screen shots of the issue as well, would love to work on it.

**Comment by AliPythonDev** [comment]
> can you attach, some screen shots of the issue as well, would love to work on it.

I have attached two screenshots for clarity:
1. In the first screenshot, you can see that I added a guest with an email address from Pakistan (my personal email), but before booking the meeting, I changed the timezone to India.
2. In the second screenshot, you can see that the guest received the meeting invite with the same timezone that I selected (India), even though the guest is from Pakistan.

This shows the problem where the guest’s local timezone is not shown correctly. This can cause confusion when planning meetings with people from different countries.

If you need more information or have any questions, just let me know. I’d love to help with this feature to make things better!
![Screenshot-I](https://github.com/user-attachments/assets/5444722c-5fc6-4c5d-87d2-00792de2f912)
![Screenshot-II](https://github.com/user-attachments/assets/11c0125e-32b7-4e00-957a-efeac4df1b78)

**Comment by ShreyTanna29** [comment]
hey @AliPythonDev ,

I think when an international guest is added, first an auto detect of timezone should happen according to the country guest is in and show that time zone conversion the guest only and second, guest should also get an option to convert the timezone in which he/she likes it. If we ask organizer to change timezone according to guest, it might be an issue when multiple guests are invited from different time zones, organizer might get confused which timezone to add in this case, what do you think about this solution ?

**Comment by Praashh** [comment]
> hey @AliPythonDev ,
> 
> I think when an international guest is added, first an auto detect of timezone should happen according to the country guest is in and show that time zone conversion the guest only and second, guest should also get an option to convert the timezone in which he/she likes it, in the cal.com web app itself. If we ask organizer to change timezone according to guest, it might be an issue when multiple guests are invited from different time zones, organizer might get confused which timezone to add in this case, what do you think about this solution ?

how will you detect the guest?  considering we are just putting the mail

**Comment by ShreyTanna29** [comment]
> > hey @AliPythonDev ,
> > I think when an international guest is added, first an auto detect of timezone should happen according to the country guest is in and show that time zone conversion the guest only and second, guest should also get an option to convert the timezone in which he/she likes it, in the cal.com web app itself. If we ask organizer to change timezone according to guest, it might be an issue when multiple guests are invited from different time zones, organizer might get confused which timezone to add in this case, what do you think about this solution ?
> 
> how will you detect the guest? considering we are just putting the mail

if the guest is cal com user than only it is possible to show them automated time zones and give options.

**Comment by AliPythonDev** [comment]
> > > hey @AliPythonDev ,
> > > I think when an international guest is added, first an auto detect of timezone should happen according to the country guest is in and show that time zone conversion the guest only and second, guest should also get an option to convert the timezone in which he/she likes it, in the cal.com web app itself. If we ask organizer to change timezone according to guest, it might be an issue when multiple guests are invited from different time zones, organizer might get confused which timezone to add in this case, what do you think about this solution ?
> > 
> > 
> > how will you detect the guest? considering we are just putting the mail
> 
> if the guest is cal com user than only it is possible to show them automated time zones and give options.

Even if the guest is a Cal.com user, the same problem still occurs. I have tried it, and it doesn’t correctly show their automated timezone.

I believe adding a feature to manually select or confirm timezones for guests would be beneficial for everyone involved.

**Comment by Praashh** [comment]
> > > hey @AliPythonDev ,
> > > I think when an international guest is added, first an auto detect of timezone should happen according to the country guest is in and show that time zone conversion the guest only and second, guest should also get an option to convert the timezone in which he/she likes it, in the cal.com web app itself. If we ask organizer to change timezone according to guest, it might be an issue when multiple guests are invited from different time zones, organizer might get confused which timezone to add in this case, what do you think about this solution ?
> > 
> > 
> > how will you detect the guest? considering we are just putting the mail
> 
> if the guest is cal com user than only it is possible to show them automated time zones and give options.

yes exactly but it does not apply to every guest, The guest may be a user of cal.com or may not be. having a timezone that the user can set is a solution.

**Comment by ShreyTanna29** [comment]
hey @Praashh , are you working on this ?, if not then I would love to work on this issue.

**Comment by Praashh** [comment]
> hey @Praashh , are you working on this ?, if not then I would love to work on this issue.

go ahead.

**Comment by ShreyTanna29** [comment]
hey @AliPythonDev ,  right now setting different time zones for all guests together feature is already available in the cal.com. for test purpose I created an meeting and set timezone to america/denver, to me (host) it shows IST as I am from India, but to guests whom i have added sees america/denver, so the feature you are asking for already exists, do want to set different time zones for each and every guests ?

**1.) screen shown to me (host).**


![thisshowntome](https://github.com/user-attachments/assets/3bf3f45e-0ec8-4c79-af3d-253d29a63397)


**2.) guest's email.**


![phone](https://github.com/user-attachments/assets/af3e938d-43c0-4f77-bf88-1c228339f5ca)



**3.) time zone set on this screen applies to guest only and not to host.**


![eventpreview](https://github.com/user-attachments/assets/5d14cd0e-fbf0-4998-98e2-4efbedfe4d0f)

**Comment by AliPythonDev** [comment]
@ShreyTanna29 There’s still an issue: when I added a guest from Pakistan and changed the timezone to India, the guest received the invite with the same timezone I selected. This means the guest’s local timezone isn’t displayed correctly, which can cause confusion when scheduling meetings with participants from different countries.

**Comment by ShreyTanna29** [comment]
> @ShreyTanna29 There’s still an issue: when I added a guest from Pakistan and changed the timezone to India, the guest received the invite with the same timezone I selected. This means the guest’s local timezone isn’t displayed correctly, which can cause confusion when scheduling meetings with participants from different countries.

Ok, then let me add functionality for host to individually select time zone for every guests so the guest can get their local time zone.

**Comment by Praashh** [comment]
> > @ShreyTanna29 There’s still an issue: when I added a guest from Pakistan and changed the timezone to India, the guest received the invite with the same timezone I selected. This means the guest’s local timezone isn’t displayed correctly, which can cause confusion when scheduling meetings with participants from different countries.
> 
> Ok, then let me add functionality for host to individually select time zone for every guests so the guest can get their local time zone.

if a user can select the guest's timezone, that will resolve it.

**Comment by Praashh** [comment]
Hey @ShreyTanna29 , are you working on it?

**Comment by ShreyTanna29** [comment]
> Hey @ShreyTanna29 , are you working on it?

Yes, from past 3 I was not feeling well and was hospitalized, but now I am good and will go ahead to solve this.

**Comment by ShreyTanna29** [comment]
hey @Praashh @AliPythonDev ,

I am stuck on an issue from last 3 hours, please help me out, 

first, in order to add timezones for guests I opened the component which rendered booker form and after a bit of searching I opened "getBookingFields.ts" at "packages/features/bookings/libs", from these file booking fields are sent to form to get rendered, In order to add timezone field with add_guests field , we need to make change in its type "multiEmail" so I opened "MultiEmail.tsx" at "packages/ui/form", which had the values/fields shown in booker form, but  when i tried to make changes on it, it showed no effect on main form, after searching more I also checked "AddGuestsDialog.tsx" at "apps/web/component/dialog" but making changes in it also has no effect on main booking form fields, I also tried multiple ways and checked multiple files but cant find that from where these fields can be modified, so please help me to find out that from exactly which files, fields in booking form can be modified.

**Comment by CarinaWolli** [comment]
This is not planned at the moment. We can revisit if this comes up again 🙏
